### ISSUE TITLE
run_qgis() fails silently when OUTPUT not specified and load_output = TRUE

### ISSUE THREAD

**Issue by Robinlovelace** [jannes-m/RQGIS#109]
Reproducible example:

``` r
library(spData)
library(sf)
#> Linking to GEOS 3.5.1, GDAL 2.2.2, proj.4 4.9.2
library(RQGIS)
#> Loading required package: reticulate

poly_1 = st_buffer(lnd[1, ], dist = 2)
#> Warning in st_buffer.sfc(st_geometry(x), dist, nQuadSegs): st_buffer does
#> not correctly buffer longitude/latitude data
#> dist is assumed to be in decimal degrees (arc_degrees).
poly_2 = world[grepl("United K", world$name_long), ]
set_env()
#> Assuming that your root path is '/usr'!
#> $root
#> [1] "/usr"
#> 
#> $qgis_prefix_path
#> [1] "/usr/bin/qgis"
#> 
#> $python_plugins
#> [1] "/usr/share/qgis/python/plugins"
int = run_qgis("qgis:intersection",
               INPUT = poly_1,
               INPUT2 = poly_2,
               OUTPUT = file.path(tempdir(), "out.shp"),
               load_output = TRUE)
#> $OUTPUT
#> [1] "/tmp/RtmpG2itSw/out.shp"
int
#> Simple feature collection with 1 feature and 17 fields
#> geometry type:  POLYGON
#> dimension:      XY
#> bbox:           xmin: -2.330738 ymin: 50.55139 xmax: 1.584034 ymax: 53.43704
#> epsg (SRID):    4326
#> proj4string:    +proj=longlat +datum=WGS84 +no_defs
#> # A tibble: 1 x 18
#>   NAME  GSS_CODE HECTARES NONLD_AREA ONS_INNER SUB_2009 SUB_2006 iso_a2
#>   <chr> <chr>       <dbl>      <dbl> <chr>     <chr>    <chr>    <chr> 
#> 1 King… E090000…    3726.          0 F         <NA>     <NA>     GB    
#> # ... with 10 more variables: name_long <chr>, continent <chr>,
#> #   region_un <chr>, subregion <chr>, type <chr>, area_km2 <dbl>,
#> #   pop <dbl>, lifeExp <dbl>, gdpPercap <dbl>, geometry <POLYGON [°]>
int = run_qgis("qgis:intersection",
               INPUT = poly_1,
               INPUT2 = poly_2,
               load_output = TRUE)
#> $OUTPUT
#> [1] "/tmp/processing5549382d8393424f899765a73a2cb9b6/a61fb54f14c24444af72ebf6dc4adddd/OUTPUT.shp"
int
#> named list()
```

Created on 2018-08-22 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by Robinlovelace** [comment]
It should at least provide a warning that `int` was not outputed as intended. Ideally `int` would be the same after the second `run_qgis()`.

**Comment by Robinlovelace** [comment]
Not such a minimal example:


``` r
library(spData)
library(sf)
#> Linking to GEOS 3.5.1, GDAL 2.2.2, proj.4 4.9.2
library(RQGIS)
#> Loading required package: reticulate

poly_1 = st_buffer(lnd[1, ], dist = 2)
#> Warning in st_buffer.sfc(st_geometry(x), dist, nQuadSegs): st_buffer does
#> not correctly buffer longitude/latitude data
#> dist is assumed to be in decimal degrees (arc_degrees).
poly_2 = world[grepl("United K", world$name_long), ]
plot(st_geometry(poly_1))
plot(poly_2, add = TRUE)
#> Warning in plot.sf(poly_2, add = TRUE): ignoring all but the first
#> attribute
system.time({
  int_sf = st_intersection(poly_1, poly_2)
})
#> although coordinates are longitude/latitude, st_intersection assumes that they are planar
#> Warning: attribute variables are assumed to be spatially constant
#> throughout all geometries
#>    user  system elapsed 
#>   0.012   0.000   0.012
plot(int_sf, add = TRUE)
#> Warning in plot.sf(int_sf, add = TRUE): ignoring all but the first
#> attribute


set_env()
#> Assuming that your root path is '/usr'!
#> $root
#> [1] "/usr"
#> 
#> $qgis_prefix_path
#> [1] "/usr/bin/qgis"
#> 
#> $python_plugins
#> [1] "/usr/share/qgis/python/plugins"
# open_app()
length(algs)
#> Error in eval(expr, envir, enclos): object 'algs' not found
# using R named arguments#
system.time({
  int = run_qgis("qgis:intersection",
                 INPUT = poly_1,
                 INPUT2 = poly_2,
                 OUTPUT = file.path(tempdir(), "out.shp"),
                 load_output = TRUE)
})
#> $OUTPUT
#> [1] "/tmp/Rtmp7ATb2l/out.shp"
#>    user  system elapsed 
#>   2.756   0.580  63.389
int = run_qgis("qgis:intersection",
               INPUT = poly_1,
               INPUT2 = poly_2,
               OUTPUT = file.path(tempdir(), "out.shp"),
               load_output = TRUE)
#> $OUTPUT
#> [1] "/tmp/Rtmp7ATb2l/out.shp"
plot(int, add = TRUE)
#> Warning in plot.sf(int, add = TRUE): ignoring all but the first attribute
```

![](https://i.imgur.com/f17MrmF.png)

``` r

identical(st_geometry(int), st_geometry(int_sf))
#> [1] FALSE

int_weird = run_qgis("qgis:intersection", 
               INPUT = poly_1,
               INPUT2 = poly_2,
               load_output = TRUE)
#> $OUTPUT
#> [1] "/tmp/processing00a86755144441bcba124b70a4062722/806ab470db9f4f54a01c71780c39b5a6/OUTPUT.shp"

plot(int_weird)
#> Error in xy.coords(x, y, xlabel, ylabel, log): 'x' is a list, but does not have components 'x' and 'y'
int_weird
#> named list()
```

Created on 2018-08-22 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by jannes-m** [comment]
Robin, you have not specified any output file. If you don't do that, QGIS will save the output to a temporary folder specified by QGIS. In fact, `run_qgis()` is even telling you where you can find the file `"/tmp/processing00a86755144441bcba124b70a4062722/806ab470db9f4f54a01c71780c39b5a6/OUTPUT.shp"`. But in the `run_qgis()` documentation you will find that `load_output` only works when the user has specified an output object. The reason behind this is that some algorithms such as `rgrass7:r.slope.aspect` generate several output files and usually the user is not interested in all of them but only in those which s/he specified. Still, one could add a warning saying `load_output` only works when the user has specified at least one output location.

**Comment by jannes-m** [comment]
Closed via c4a5e40
